### PR TITLE
gh-action: add peerpod controller workflow

### PR DESCRIPTION
**Pull request by Amulyam24**
Add a new github workflow to run required checks for peerpod controller

This PR depend on: #675 

Fixes: #665

### CODE REVIEW

**Comment by Amulyam24** [comment]
Tested the workflow locally using this tool - https://github.com/nektos/act

File: .github/workflows/controller.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  controller_job:
+    name: controller
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: peer-pod-controller
+    strategy:
+      matrix:
+        go_version:
+          - 1.18
```
Comment (Amulyam24) [inline .github/workflows/controller.yaml:18]: Currently golang version is 1.18, will need update once this PR is merged - #648

**Comment by snir911** [comment]
Thanks!
Is it failing because of the go version?
LGTM overall, nit, there's minor typo in the subject s/coontroller/controller

**Comment by snir911** [comment]
oh, just noticed it's failing because of #675, I think worth merging the PRs, not critical though..

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @Amulyam24

**Review by snir911 — APPROVED** [review]
thanks!

**Pull request merged**